File: BE/PrePay/src/main/java/com/d111/PrePay/dto/respond/TeamDetailRes.java
package com.d111.PrePay.dto.respond;


import com.d111.PrePay.model.Team;
import com.d111.PrePay.model.UserTeam;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter
@AllArgsConstructor
@Builder
public class TeamDetailRes {
    private Long teamId;
    private String teamName;
    private int dailyPriceLimit;
    private boolean publicTeam;
    private int countLimit;
    private boolean position;
    private String teamMessage;
    private String teamPassword;

    public TeamDetailRes(Team team, UserTeam userTeam){
        this.teamId = team.getId();
        this.teamName = team.getTeamName();
        this.dailyPriceLimit = team.getDailyPriceLimit();
        this.publicTeam = team.isPublicTeam();
        this.teamPassword = team.getTeamPassword();
        this.countLimit = team.getCountLimit();
        this.position =  userTeam.isPosition();
        this.teamMessage = team.getTeamMessage();
    }
}

File: Prepay/app/src/main/java/com/example/prepay/data/response/TeamDetailRes.kt
package com.example.prepay.data.response

data class TeamDetailRes(
    val countLimit: Int,
    val dailyPriceLimit: Int,
    val position: Boolean,
    val publicTeam: Boolean,
    val teamId: Int,
    val usedAmount : Int,
    val teamMessage: String,
    val teamName: String,
    val teamPassword: Any,
    val color: String,
    val teamBalance: Int
)